### PR TITLE
Feature/cfp frontend

### PR DESCRIPTION
**Pull request by estyxx**
Create CFP page 

TO-DO
- [x] Remove yup, use react-use-form-state?
- [x] Error messages
- [x] Get SubmissionType from BE **#helpwanted** **#seeSlack :) **
- [x] WIP Update mutation
- [x] Look at `Select a valid choice. That choice is not one of the available choices.` Mutation Error
- [ ] Check user is authenticated -> done in #1198 only need to merge it and use PrivateRoute for CfpForm also :)
- [x] When a SubmissionType is selected Check options Duration!rm AT ALL**
- [ ] Fix sendSubmission variables "conference": pass CONFERENCE_CODE of the enviroment! Now it's hardcoded... I have to figure out how to pass it as gatsby does... I have to do the same with apollo. **#helpwanted**
- [x] When the SendSubmissionMutation is successful Show Alert 
- [ ] ~When the SendSubmissionMutation is successful **do not permit to edit the form anymore?**~ For now not. 
- [ ] Try [separate branch] https://react-hook-form.com/ [see issue](https://github.com/pythonitalia/pycon/issues/1186)
- [x]  Use FormattedMassege properties to interpolate the Deadline date in the Alert (Now if there isn't any Deadline the message is truncated "Sorry, bla bla bla, CFP will open from" (without any date)... avoid "CFP will open from" if you don't know when... 
```
<FormattedMessage
  id="app.greeting"
  description="Greeting to welcome the user to the app"
  defaultMessage="Hello, {name}!"
  values={{
    name: 'Eric',
  }}
/>
```
- [x] elevatorPitch: yup: string.max 300!
- [x] Maybe is better to take the SubmissionType from the duration (and remove the Type added to the conference). 
```
query($code: String!) {
            conference(code: $code) {
                durations {
                    allowedSubmissionTypes {
                        id
                        name
                    }
                }
            }
        }
```

### CODE REVIEW

File: frontend/src/routes/cfp/form.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+export const CFPForm = () => {
+  const [errors, setErrors] = useState({});
+  const [submission, setSubmission] = useState({
+    title: "",
+    abstract: "",
+    elevatorPitch: "",
+    topic: "",
+    duration: "",
+    language: "eng",
```
Comment (marcoacierno) [inline frontend/src/routes/cfp/form.tsx]: The default value is a language is `en` if I remember correctly

Comment (estyxx) [reply frontend/src/routes/cfp/form.tsx]: Yes, but here I am working with sample data, now I think I can do a Query with gatsby to retrieve constants, right??

**Comment by estyxx** [comment]
Hi!

I think this PR starts to be complete :) 

The **main point** left is
> Fix sendSubmission variables `code`: pass CONFERENCE_CODE from the enviroment! Now it's hardcoded... 

For this, **I need help @marcoacierno @fedriz tnks :)**

This point: 
>  When the SendSubmissionMutation is successful do not permit to edit the form anymore?

I don't know... for now I disabled everything after the submission I sent.

This: 
>  Try [separate branch] https://react-hook-form.com/ see issue

I'll do later o maybe in a separate PR, now I have to finish profile-edit... yup + use-form-react-state works for now... 

Tnks guys :)

**Comment by estyxx** [comment]
P. S. Every check is failed... I don't understand very much... I didn't modify much in the BE, for the FE I'm running pre-commit, so every tslint-issue should be already fixed...

**Pull request closed without merging**